reactive/zookeeper.py

```diff
@@ -0,0 +1,51 @@
+'''
+Reactive handlers for Zookeeper.
+'''
+
+from charmhelpers.core import hookenv
+from charms.layer.zookeeper import Zookeeper
+from charms.reactive import set_state, when, when_not
+
+@when('bigtop.available')
+@when_not('zookeeper.installed')
+def install_zookeeper():
+    '''
+    After Bigtop has done the initial setup, trigger a puppet install,
+    via our Zooekeeper library.
+
+    puppet will start the service, as a side effect.
+
+    '''
+    zookeeper = Zookeeper()
+    zookeeper.install()
+    zookeeper.open_ports()
+    set_state('zookeeper.installed')
+    set_state('zookeeper.started')
+    hookenv.status_set('active', 'Ready')
+
+@when('zookeeper.started', 'zkpeer.joined')
+def quorum_add(zkpeer):
+    """Add a zookeeper peer.
+
+    Add the unit that just joined, restart Zookeeper, and remove the
+    '.joined' state so we don't fall in here again (until another peer joins).
+    """
+    nodes = zkpeer.get_nodes()  # single node since we dismiss .joined below
```

Comment:
Same as above. Only no you would need to do:

"hookenv.status_set('waiting', 'Configuring Zookeeper')"
Then you would do the install calls that take a considerable amount of time (calls that include puppet apply)
And at the end you set the state to Ready again:
"hookenv.status_set('active', 'Ready')"